Add a manifest for Sony devices

stellirin opened the pull request:
Sony provides device repos that can build on pure AOSP. These can be used on Copperhead OS unmodified, You just need to include the proprietary blobs and a couple of small cherry-picks to enable some AOSP repos for Sony devices.

**For the complete experience you'll need:**
1. Git repo for Sony proprietary blobs
2. Git repo for Modem blobs from older devices
3. AOSP forks for some minor Sony changes

You may also want local forks of Sony's repos for stability, however for testing and development it's enough to sync directly from Sony's GitHub, following the instructions on [Sony's Website](http://developer.sonymobile.com/knowledge-base/open-source/open-devices/aosp-build-instructions/how-to-build-aosp-marshmallow-for-unlocked-xperia-devices/).

Tested on several devices, working as expected.

stellirin committed:
Add a manifest for Sony devices

Sony provides deviec repos that can build on pure AOSP. These can
be used on Copperhead OS unmodified, You just need to include the
proprietary blobs and a couple of small patches to enable some AOSP
repos for Sony devices.

thestinger commented:
I appreciate the effort in setting this up and testing it, but I don't want to include this.

I'm not sure there's a benefit to bundling more repositories for expanded device support. It will greatly increase the amount of downloaded code, and we aren't going to make any official builds that are untested. If people want to build this themselves, they could just use local manifests. I would just be closing any bug reports from non-officially-supported devices as invalid, so including the code would be misleading.

I'm also only interested in supporting devices meeting the security standards of the project, meaning that reasonable security updates are provided for the proprietary components. That isn't the case for most (all?) of these devices and there's no commitment from Sony to provide security updates consistently. I would also be unwilling to make any changes outside the manifest repository, if that's really required. Expanded device support would need to leave repositories used by other devices completely untouched.

Either way, all changes that we make need to be regularly and thoroughly tested, and we don't have the resources to expand to more devices. CopperheadOS aims to improve the security of the Android Open Source Project, not expand the device support it offers. It just isn't within the scope of the project. We have a lot of trouble simply supporting the current set of devices, let alone expanding beyond the full Nexus line (which isn't even completely covered yet).

thestinger commented:
For just one example of the need for thorough, device-specific testing, some apps like Firefox are likely going to crash with these builds since CopperheadOS has a hard requirement on custom kernels with extra backports. I'm sure there are also going to be memory corruption bugs uncovered by OpenBSD malloc. And someone is going to need to port PaX to each kernel... It's just not realistic. The only way to maintain high quality standards is limiting the scope of the project.

stellirin commented:
I put the sony repos in the `sony.xml` specifically so that it doesn't interfere with the nexus devices, you have to specify `-m sony.xml` when doing `repo init` which addresses some of your concerns.

Part of the beauty of Sony's Open Devices project is that there is one single 3.10 based kernel for all devices, so PaX and any other features need to be ported only once (looking at the bullhead kernel there doesn't seem to be too much backporting so far).

You're right (AFAIK) that there's no real guarantee from Sony specifically over security updates for the proprietary components. However the proprietary blobs provided for the Open Devices project are compiled from the latest QCOM code for Android 6.0 Marshmallow, so any and all bug fixes _should_ be included.[1] As can be seen [here](http://developer.sonymobile.com/knowledge-base/open-source/open-devices/latest-updates/) the blobs are recompiled every month or so.

If you don't want to merge this here, would you allow us to (later) publish unofficial builds at [fxpblog.co](http://www.fxpblog.co) using the Copperhead OS name, if we determine there's a big enough demand for it?

[1] I say should, as that will need confirmation for specific known bugs, but it's definitely something I can find out.

thestinger commented:
> You're right (AFAIK) that there's no real guarantee from Sony specifically over security updates for the proprietary components. However the proprietary blobs provided for the Open Devices project are compiled from the latest QCOM code for Android 6.0 Marshmallow, so any and all bug fixes should be included.[1] As can be seen here the blobs are recompiled every month or so.

It's good that they're releasing updates, but I need more than that. I need to know that they're actually doing real security updates, including for the radio and non-Qualcomm proprietary code.

> If you don't want to merge this here, would you allow us to (later) publish unofficial builds at fxpblog.co using the Copperhead OS name, if we determine there's a big enough demand for it?

If I did merge this, it wouldn't make builds available. I don't have the resources to do any more builds, and any official releases would need to be tested - which means buying devices, and spending time that I don't have available. There are going to be compatibility issues that come up, and those need to be dealt with too (narrowed down, debugged, fixed / worked around / reported)

There's no problem with people porting the OS to other devices and publishing builds, but it needs to be clear that it's not official and that there are caveats. I don't think that it would make sense for it to be presented as CopperheadOS, but stating that it's an unofficial port of CopperheadOS would be fine - similar to how CopperheadOS refers to PaX in the documentation at https://copperhead.co/android/.

thestinger commented:
This is currently out-of-scope, but it can be considered in the future when the project has substantially more resources than today (i.e. not just a single not-quite-full-time developer).
